testplan/runners/pools/base.py

```diff
@@ -643,11 +657,14 @@ def workers_requests(self):
     def stopping(self):
         """Stop connections and workers."""
         # TODO do we need a lock here?
-        super(Pool, self).stopping()  # stop the loop and the monitor
 
+        self._workers.stop()
         for worker in self._workers:
             worker.transport.disconnect()
-        self._workers.stop()
+
+        self._exit_loop = True
```

Comment:
set _exit_loop after all workers are stopped.
We cannot use pool.status.tag == stopping for exiting loop because threadmill workers needs the loop running til worker stopped.